docs/technical_tutorial/apollo_vehicle_adaption_tutorial_cn.md

```diff
@@ -0,0 +1,935 @@
+
+![gear1](images/vehicle_adaption_tutorial/gear1.jpg)
+##### 5. 反馈信号和控制信号中如车速，轮速，加速度，踏板位置（百分比）等double类型的反馈和控制信号在DBC中Value Description项中必须为空。
+对于实时数值反馈信号和数值控制信号，如车速（实际车速）、轮速反馈（实际轮速），踏板控制（百分比），转角控制（实际转角值）等，此类信号在定义Value Description项中不能加任何内容。
+
+![val](images/vehicle_adaption_tutorial/val.jpg)
+##### 6. 转向信号的范围，在定义时要填写准确的取值范围，注意控制转角的精度一般不高于0.05deg，踏板百分比精度（factor）不高于0.1。
+对于所有报文的Byte Order，一个DBC内的信号只能统一定义，全部是Motorola格式或者全部是Intel格式。
+
+![define](images/vehicle_adaption_tutorial/define.jpg)
+
+
+## 二、适配CANBUS代码
+### 1. DBC文件转换成canbus模板代码
+Canbus适配代码可以使用apollo的工具生成，在转换代码前，要保证DBC按照上述的DBC文件要求完成，并通过gedit打开dbc文件，另存转码为UTF-8格式保存。
+
+（1）将DBC文件放置指定目录下，目录apollo/modules/tools/gen_vehicle_protocol内。
+
+（2）修改DBC转换脚本的配置文件：下面以GE3车型添加为例，在apollo/modules/tools/gen_vehicle_protocol目录下，复制默认存在的mkz_conf.yml文件并重命名为ge3_conf.yml，修改该配置文件，如下图所示：
+![yml](images/vehicle_adaption_tutorial/yml.png)
+dbc_file：填写对应你的DBC文件名称，DBC文件名称一般以车型名称命名，并以.dbc结束；
+
+protocol_conf：以你DBC文件名称命名，如ge3.dbc，则这里命名为ge3.yml，同时将该配置文件名称修改为ge3_conf.yml；
+
+car_type：填入你的车型名称；
+
+sender_list：[ ]  ——发送列表，这里默认为空；
+
+sender：此处修改为与你在DBC内定义的Apollo控制器名称一致，在前面已经DBC要求中，建议定义Apollo的控制器名称为ACU。
+
+（3）完成ge3_conf.yml配置文件设置，启动docker，进入Apollo的容器后，在apollo/modules/tools/gen_vehicle_protocol目录下，找到DBC转化工具gen.py，执行代码：
+
+    cd modules/tools/gen_vehicle_protocol
+    python gen.py ge3_conf.ymal
+执行完成上述脚本后，在终端内会显示生成了控制协议5个，反馈协议11个。
+![creat_ge3protocol](images/vehicle_adaption_tutorial/creat_ge3protocol.png)
+这时在apollo/modules/tools/gen_vehicle_protocol目录下，会生成一个output文件夹，文件夹内有2个文件夹，一个是proto文件夹，一个是vehicle文件夹；这两个文件内的代码内容就是我们要适配canbus的基本代码模板了。我们需要把文件内的代码**拷贝**到apollo的canbus层内，进行代码适配添加。
+![output](images/vehicle_adaption_tutorial/output.png)
+
+**注意：**把这个output文件夹内生成的代码模板拷贝至相应的apollo目录后，要删除该文件夹，如果不删除该文件夹，后期编译apollo时会报错。该文件夹有保护权限，请在apollo的docker内执行删除代码：
+
+    rm -rf output/
+
+### 2. 适配代码合入apollo文件内
+下面以添加ge3车型为例，将该代码添加至apollo内：
+![output](images/vehicle_adaption_tutorial/output.png)
+
+（1）将apollo/modules/tools/gen_vehicle_protocol/output/proto 文件夹内ge3.proto文件拷贝至apollo/modules/canbus/proto 文件夹内，并在该文件夹内修改chassis_detail.proto，在该文件头部添加头文件`import "modules/canbus/proto/ge3.proto"`
+![importproto](images/vehicle_adaption_tutorial/importproto.png)
+在message ChassisDetail{} 结构体内的最后一行添加要增加的新车型变量定义： `Ge3 ge3 = 21`；
+![ge3_chassis_detail](images/vehicle_adaption_tutorial/ge3_chassis_detail.png) 
+在pollo/modules/canbus/proto目录的BUILD文件内添加上述新proto的依赖：`"ge3.proto"`；
+![BUILD_depend](images/vehicle_adaption_tutorial/proto_BUILD_depend.png)
+（2）将apollo/modules/tools/gen_vehicle_protocol/output/vehicle/ 内的ge3文件夹拷贝至apollo/modules/canbus/vehicle/ 文件夹下；
+![copy_ge3](images/vehicle_adaption_tutorial/copy_ge3.png)
+### 3.实现新的车辆控制逻辑
+实现新的车辆控制逻辑，在apollo/modules/canbus/vehicle/ ge3 ge3_controller.cc 文件编写控制逻辑代码，主要包含将解析的底盘反馈报文的信息，通过chassis和chassis_detail channel广播出来供其它模块使用底盘信息。chassis主要包括获取底盘的车速、轮速、发动机转速、踏板反馈、转角反馈等信息， chassis_detail是每一帧报文的实际信息，这一部分编写的代码如下所示：
+
+    // 3
```

Comment:
Double check and make sure those are in code format in markdown. refer to https://help.github.com/en/articles/creating-and-highlighting-code-blocks if needed